.github/workflows/push-adkg-cli-docker.yml

```diff
@@ -62,16 +70,44 @@ jobs:
             image=mirror.gcr.io/moby/buildkit:buildx-stable-1
             network=host
 
+      - name : Add SHORT_SHA Environment Variable
+        id   : short-sha
+        shell: bash
+        run  : echo "SHORT_SHA=`git rev-parse --short HEAD`" >> $GITHUB_ENV
+
       - id: docker-push-tagged
         name: Tag Docker image and push to Google Artifact Registry
         uses: docker/build-push-action@v6
         with:
           push: ${{ github.ref == 'refs/heads/main' }}
-          tags: ${{ steps.meta.outputs.tags }}
+          tags: |
+                ${{ steps.meta.outputs.tags }}
+                ghcr.io/randa-mu/dcipher/adkg-cli:main-latest
+                ghcr.io/randa-mu/dcipher/adkg-cli:${{ env.SHORT_SHA }}
           labels: ${{ steps.meta.outputs.labels }}
           context: .
           file: ./crates/adkg-cli/Dockerfile
           cache-from: |
             type=registry,ref=${{ env.DOCKER_REGISTRY }}/adkg-cli-cache:${{ steps.meta.outputs.version }}
             type=registry,ref=${{ env.DOCKER_REGISTRY }}/adkg-cli-cache:main-latest
           cache-to: type=registry,ref=${{ env.DOCKER_REGISTRY }}/adkg-cli-cache:${{ steps.meta.outputs.version }},mode=max
+
+      - name : extract adkg-cli binary
+        shell: bash
+        run  : |
+          docker pull ghcr.io/randa-mu/dcipher/adkg-cli:${{ env.SHORT_SHA }}
+          container=$(docker create ghcr.io/randa-mu/dcipher/adkg-cli:${{ env.SHORT_SHA }})
+          docker cp $container:usr/local/bin/adkg-cli ./adkg-cli
+          docker rm $container
+
+      - name: Upload adkg-cli binary as artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: adkg-cli-binary
+          path: ./adkg-cli
+
+      - name: upload binaries to release
+        uses: softprops/action-gh-release@v1
```

Comment:
Maybe just use the `gh release` CLI instead of a random action from `softprops` to avoid supply chain risks

Reply:
done!